File: LaGranjaSA/src/main/java/com/example/LaGranjaSA/modelo/Alimentacion.java
package com.example.LaGranjaSA.modelo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "alimentacion")
public class Alimentacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "fk_id_raza")
    private Raza raza;
    private String etapa;
    private String descripcion;
    private float dosis;
}

File: LaGranjaSAFront/LaGranjaFront/src/app/model/Alimentacion.ts
import { Raza } from "./Raza";


export class Alimentacion {
    constructor(
        id: number,
        etapa: String,
        descripcion: String,
        dosis: number,
        raza: Raza
    ) {
        this.id = id
        this.etapa = etapa
        this.descripcion = descripcion
        this.dosis = dosis
        this.raza = raza
    } public id: number;
    public etapa: String;
    public descripcion: String;
    public dosis: number;
    public raza: Raza;


}